src/betterproto/__init__.py

```diff
@@ -691,17 +704,18 @@ def __repr__(self) -> str:
     if not TYPE_CHECKING:
 
         def __getattribute__(self, name: str) -> Any:
-            """
-            Lazily initialize default values to avoid infinite recursion for recursive
-            message types
-            """
-            value = super().__getattribute__(name)
-            if value is not PLACEHOLDER:
-                return value
+            # Raise `AttributeError` on attempts to access unset `oneof` fields
+            try:
+                group_current = super().__getattribute__("_group_current")
+            except AttributeError:
+                pass
+            else:
+                if name not in ("__class__", "_betterproto"):
```

Comment:
You may want to change this to a set to get slightly better performance:

```suggestion
                if name not in {"__class__", "_betterproto"}:
```

```
$ python -m timeit '"abc" in ("def", "abc")'
10000000 loops, best of 5: 21.7 nsec per loop
$ python -m timeit '"abc" in {"def", "abc"}'
20000000 loops, best of 5: 13.5 nsec per loop```

Reply:
I've amended the change to use `{"__class__", "_betterproto"}`.